### ISSUE TITLE
False positive for `use_self` when using core::mem::transmute in From impl

### ISSUE THREAD

**Issue by SamokhinIlya** [rust-lang/rust-clippy#4734]
clippy 0.0.212 (3aea8603 2019-09-03)

```rust
#![warn(clippy::use_self)]

#[repr(C, packed)]
pub struct X {
    pub x: u32,
}

impl From<X> for u32 {
    fn from(c: X) -> Self {
        unsafe {
            core::mem::transmute(c)
        }
    }
}
```

[playground link](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=8c102939769eb74e574397e5d469c3f1)
